### DISCUSSION TITLE
Ascertain who won the round

### DISCUSSION THREAD

**Discussion by stnokott** [redraskal/r6-dissect#20] [Q&A] [answered]
Hey there!

As a hobby sideproject, I'm currently developing a UI for R6 match replays using your library as part of the backend.

I am currently trying to figure out a way to find out who won a round based on the output the library provides.
The ideal result would be a boolean indicating if the team of the observing player won the round.

With this as a usecase, I struggle to find a reliable solution.
My ideas were these two:

1. Process the score after a round to determine which team won it, then derive from that if the observing player was part of that team.
In theory, this sounded like the simplest approach. It wouldn't work obviously for custom games where the observer doesn't actually participate, but for my usecase, this scenario would never occur, so no issue on that side.
What did prove a challenge though was to map the team with the winning score in `dissect.Header.Teams` to a player:
The team names ("OPPONENT", "YOUR TEAM") didn't seem reliable enough to me to just assume that "YOUR TEAM" is the observing player's team. Using the team index from `dissect.Header.Players[].TeamIndex` also proved to not always match with the index of the corresponding team in `dissect.Header.Teams`.
Thus, this approach was deemed too unreliable to get working.

2. This idea encompassed finding the chronologically last event in the activity feed and determining from that who "finished" the round, i.e. who got the round-ending kill or who finished defusing the bomb.
This was the second, admittedly, slightly desperate approach at trying to come up with a solution for the problem at hand.
During my attempt at an implementation, I realised that the library misses a few possible round-ending events, some as simple as "time ran out".
So this solution also wasn't a success.

Thanks for reading thus far, do you have any idea how to find out if the observing player won a round?
Maybe it's not possible with the current output of the library?

Thanks a lot for this library @redraskal , I'm looking forward to any ideas!

**Comment by redraskal** [comment]
Hey @stnokott,

The two ways you describe are the current methods for determining the winner. I will add the winning team to the header whenever I figure out a reliable method for pulling that information. I'm currently working on fixing some problems with Year 8 replays and will hopefully get a solution on that soon.

~Can you send a replay where TeamIndex is not accurate?~ found the problem

**Reply by stnokott** [reply]
I can still provide you with a replay file and/or a minimal reproducible example if that helps.
Appreciate you looking into this!

**Reply by stnokott** [reply]
I can put all of this in an issue if that helps.

**Comment by redraskal** [comment, marked as answer]
The solution is likely an algorithm similar to your second idea with additional data.

**Reply by stnokott** [reply]
For now, I will just use the team with `.Name == "YOUR TEAM"` and assume that that is the team of the recording player, disregarding the `.TeamIndex` field entirely.
That should work reliably.

This does mean though that I will need data about the previous round (if not looking at the very first) to determine who won the round.

**Reply by redraskal** [reply]
> For now, I will just use the team with `.Name == "YOUR TEAM"` and assume that that is the team of the recording player, disregarding the `.TeamIndex` field entirely.
> 
> That should work reliably.
> 
> 
> 
> This does mean though that I will need data about the previous round (if not looking at the very first) to determine who won the round.

@stnokott Issues here have since been resolved on the main branch. Will need to make an adjustment to fix matches with less than 10 players. Team struct now includes win data.

Funny enough, it looks like usernames are sent in order (first 5 on team 1). This also works on pro league replays where the recording player is a spectator.

**Reply by stnokott** [reply]
> Funny enough, it looks like usernames are sent in order (first 5 on team 1). This also works on pro league replays where the recording player is a spectator.

Ah that is very nice. Looking forward to the next release!

**Reply by stnokott** [reply]
You even included the win condition, this is perfect, thanks!
